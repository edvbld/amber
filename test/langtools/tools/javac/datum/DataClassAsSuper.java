/*
 * @test /nodynamiccopyright/
 * @summary check that a datum can inherit from DataClass or an abstract datum class
 * @compile/fail/ref=DataClassAsSuper.out -XDrawDiagnostics DataClassAsSuper.java
 */

class DataClassAsSuper {

    // should extend DataClass or an abstract datum
    __datum D1(int x) extends Object { }

    __datum D2(int y) {}

    // D2 is datum but not abstract
    __datum D3(int y, int x) extends D2(y) {}
}
