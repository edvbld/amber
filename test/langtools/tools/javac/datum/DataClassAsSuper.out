DataClassAsSuper.java:10:31: compiler.err.cant.extend.datum: (compiler.misc.bad.datum.super)
DataClassAsSuper.java:15:38: compiler.err.cant.extend.datum: (compiler.misc.bad.datum.super)
2 errors
